### ISSUE TITLE
[ENHANCEMENT] Semi-annual Update and Review of Codebase runtime and dependency versions

### ISSUE THREAD

**Issue by rverma-nsl** [aws-samples/aws-secure-environment-accelerator#676]
1. Review and update ALL dependencies to latest or n-1 LTS versions
   - don't fix breaking changes, just update to determine any breaking changes
   - create a list of breaking changes and a high level assessment of effort to remediate
   - remediation will be a second ticket
   - pin all versions
2. Create a list/registry of all top-level 3rd party dependencies and license type

Dependencies to consider: (please add to this list if I missed anything)
- Node.js from 12 to 14 LTS latest
- Typescript to latest
- AWS CDK to latest
- pnpm to latest
- aws sdk to latest 
- Codebuild buildImage to latest includes:
  - codebuild.LinuxBuildImage.STANDARD_5_0
- Container image to latest
  - bitnami container - public.ecr.aws/bitnami/node:14-prod
- other 3rd party dependencies
  - prettier, ~jest~, tempy, ip-num, etc., etc.

**Comment by Brian969** [comment]
- AWS CDK has a weekly release - we have made the decision at this time to only increment the CDK release when we need a new feature, function, or fix augmented with a semi-annual dependency review cycle.  
- Nodejs 12 was the latest LTS version of Nodejs at the time we selected it, and continues to have more than a year of support.  Yes, we have plans to explore upgrading to latest Nodejs 14, but given current support status, it is not a priority over adding the functionality being demanded by our customers (including yourself :-) ).
